### PR TITLE
Feature/4621 get supplier section implementation

### PR DESCRIPTION
**Pull request by sam-roberts-bjss**
Story: [AB#4621](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4621)
Task: [AB#7180](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7180)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Api/Properties/launchSettings.json [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development",
         "ConnectionStrings__OrderingDb": "Data Source=localhost,1552;Initial Catalog=CatalogueOrdering;MultipleActiveResultSets=true;User ID=NHSD;Password=DisruptTheMarket1!"
       },
-      "applicationUrl": "http://localhost:5104"
+      "applicationUrl": "http://localhost:8076"
     }
   }
-}
+}
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Ordering.Api/Properties/launchSettings.json:22]: Add a newline at the end of the file

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary 5321.feature [hunk 2]
```diff
@@ -8,6 +8,7 @@ Background:
         | OrderId    | Description   | OrderStatusId | Created    | LastUpdated | LastUpdatedBy                        | OrganisationId                       |
         | C000014-01 | A Description | 1             | 05/05/2020 | 09/05/2020  | a11a46f9-ce6f-448a-95c2-fde6e61c804a | 4af62b99-638c-4247-875e-965239cd0c48 |
     And the user is logged in with the Buyer role for organisation 4af62b99-638c-4247-875e-965239cd0c48
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary 5321.feature:11]: Seen as you've edited this, you may as well remove the 5321 in the filename, as per disussed on friday in the code standards meeting

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Orders GetAll 4826.feature [hunk 3]
```diff
@@ -16,9 +16,9 @@ Scenario: 1. Get all of the orders from an existing organisationId
     When a GET request is made for a list of orders with organisationId 4af62b99-638c-4247-875e-965239cd0c48
     Then a response with status code 200 is returned
     And the orders list is returned with the following values
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Orders GetAll 4826.feature:18]: Seen as you've edited this, you may as well remove the 5321 in the filename, as per disussed on friday in the code standards meeting

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    Then a response with status code 404 is returned
+
+@4621
+Scenario: 3. If a user is not authorised then they cannot access the order supplier section
+    Given no user is logged in
+    When the user makes a request to retrieve the order supplier section with the ID C000014-01
+    Then a response with status code 401 is returned
+
+@4621
+Scenario: 5. A buyer user cannot access the order supplier for an organisation they don't belong to
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature:44]: You've skipped Scenario 4

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    Given the user is logged in with the Buyer role for organisation e6ea864e-ef1b-41aa-a4d5-04fc6fce0933
+    When the user makes a request to retrieve the order supplier section with the ID C000014-01
+    Then a response with status code 403 is returned
+
+@4621
+Scenario: 6. Service Failure
+    Given the call to the database will fail
+    When the user makes a request to retrieve the order supplier section with the ID C000014-01
+    Then a response with status code 500 is returned
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature:53]: Can you add a new line to the end of the feature file

File: src/NHSD.BuyingCatalogue.Ordering.Api/Properties/launchSettings.json (hunk 1 above)
Comment (sam-roberts-bjss) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Properties/launchSettings.json]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/AddressExtensionsTests.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Builders;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Extensions
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    public class AddressExtensionsTests
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/AddressExtensionsTests.cs:13]: sealed

Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/AddressExtensionsTests.cs:13]: sealed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/ContactExtensionsTests.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+using NHSD.BuyingCatalogue.Ordering.Api.Models;
+using NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Builders;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Extensions
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    public class ContactExtensionsTests
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/ContactExtensionsTests.cs:13]: sealed

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary 5321.feature (hunk 2 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Summary 5321.feature:11]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Orders GetAll 4826.feature (hunk 3 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Orders GetAll 4826.feature:18]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature (hunk 4 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature:49]: Great spot! Please don't tell me the numbering actually helped with this

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature (hunk 5 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Get.feature]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/AddressExtensionsTests.cs (hunk 6 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/AddressExtensionsTests.cs]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/ContactExtensionsTests.cs (hunk 7 above)
Comment (sam-roberts-bjss) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Extensions/ContactExtensionsTests.cs]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Orders GetAll.feature [hunk 8]
```diff
@@ -48,4 +48,4 @@ Scenario: 5. A buyer user cannot access the orders for an organisation they do n
 Scenario: 6. Service Failure
     Given the call to the database will fail
     When a GET request is made for a list of orders with organisationId 4af62b99-638c-4247-875e-965239cd0c48
-    Then a response with status code 500 is returned
+    Then a response with status code 500 is returned
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Orders GetAll.feature:51]: New line needed

**Pull request merged**